recipes/libcoap/all/conanfile.py

```diff
@@ -0,0 +1,100 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class LibCoapConan(ConanFile):
+    name = "libcoap"
+    license = "BSD-2-Clause"
+    homepage = "https://github.com/gocarlos/libcoap/"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = """A CoAP (RFC 7252) implementation in C"""
+    topics = ("coap")
+    exports_sources = "CMakeLists.txt"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "enable_dtls": [True, False],
```

Comment:
I suppose these TLS libraries are exclusive. If not, ignore this.
Check the [zeromq recipe](https://github.com/conan-io/conan-center-index/blob/1477c083de3800cec628d3f3efa8302d2cf74ed1/recipes/zeromq/all/conanfile.py#L17) how I handled this in the past.
This gives less options and avoid impossible options, such as `enable_dtls=False` and `with_openssl=True`

Reply:
Another example: [libssh2](https://github.com/conan-io/conan-center-index/blob/1477c083de3800cec628d3f3efa8302d2cf74ed1/recipes/libssh2/all/conanfile.py#L24)